apm-agent-core/src/test/java/co/elastic/apm/agent/report/serialize/MetricSetSerializationTest.java

```diff
@@ -185,11 +188,26 @@ void testServiceName() throws Exception {
         assertThat(serviceName.asText()).isEqualTo("bar");
     }
 
+    @Test
+    void testServiceNameOverride() throws Exception {
```

Comment:
Please add another test with multiple service names and one which tests that the service name overrides don't apply when `service_name` is configured.

Reply:
Should I test that `tracer.getServiceNameOverrides` returns an empty list or that the serialized metricset does not contain any service name (except in the metadata)?

I probably need a (fully) setup ElasticAPMTracer. Do you know any test that has similar requirements?

Reply:
It's probably enough to test that `tracer.getServiceNameOverrides ` is empty in `co.elastic.apm.agent.impl.ElasticApmTracerTest#testNotOverrideServiceNameWhenDefaultServiceNameConfigured`.

Reply:
Thank You.

I added a test